packages/flutter/test/widgets/sensitive_content_test.dart

```diff
@@ -0,0 +1,127 @@
+
+      await tester.pumpWidget(
+        SensitiveContent(sensitivity: ContentSensitivity.sensitive, child: childWidget),
+      );
+
+      expect(find.byWidget(childWidget), findsNothing);
+      final SizedBox shrinkBox = tester.firstWidget(find.byType(SizedBox)) as SizedBox;
+      expect(shrinkBox.width, 0);
+      expect(shrinkBox.height, 0);
+
+      setContentSensitivityCompleter.complete();
+
+      // Two pumps to complete registration, then re-build SensitiveContent widget.
+      await tester.pump();
+      await tester.pump();
+
+      expect(find.byWidget(childWidget), findsOne);
+      expect(find.byType(SizedBox), findsNothing);
+    },
+  );
+
+  testWidgets(
+    'when SensitiveContent widget changes sensitivity, SizedBox.shrink is built initially, then child widget is built upon completion',
+    (WidgetTester tester) async {
+      final Completer<void> setContentSensitivityCompleter = Completer<void>();
+      int setContentSensitivityCall = 0;
+      TestDefaultBinaryMessengerBinding.instance.defaultBinaryMessenger.setMockMethodCallHandler(
+        SystemChannels.sensitiveContent,
+        (MethodCall methodCall) {
+          if (methodCall.method == 'SensitiveContent.setContentSensitivity') {
+            setContentSensitivityCall++;
+            // Make second call to update content sensitivity awaits the Future for test.
+            if (setContentSensitivityCall == 2 && methodCall.arguments == 'autoSensitive') {
+              return setContentSensitivityCompleter.future;
+            }
+            return Future<void>.value();
+          } else if (methodCall.method == 'SensitiveContent.getContentSensitivity') {
+            return Future<String>.value(defaultContentSensitivitySetting.name);
+          } else if (methodCall.method == 'SensitiveContent.isSupported') {
+            return Future<bool>.value(true);
+          } else {
+            return null;
+          }
+        },
+      );
+
+      const Key scKey = Key('scKey');
+      final Container childWidget = Container();
+
+      await tester.pumpWidget(
+        ChangeContentSensitivityTester(
+          key: scKey,
+          initialContentSensitivity: ContentSensitivity.sensitive,
+          child: childWidget,
+        ),
+      );
+      await tester.pump();
+
+      final ChangeContentSensitivityTesterState scState = tester
+          .firstState<ChangeContentSensitivityTesterState>(find.byKey(scKey));
+      scState.changeContentSensitivityTo(ContentSensitivity.autoSensitive);
+
+      await tester.pump();
+
+      expect(find.byWidget(childWidget), findsNothing);
+      final SizedBox shrinkBox = tester.firstWidget(find.byType(SizedBox)) as SizedBox;
+      expect(shrinkBox.width, 0);
+      expect(shrinkBox.height, 0);
+
+      await tester.pump();
+
+      setContentSensitivityCompleter.complete();
+
+      // Two pumps to complete re-registration, then re-build SensitiveContent widget.
+      await tester.pump();
+      await tester.pump();
```

Comment:
Thanks for explaining this. I'm glad you ended up not needing pumpAndSettle or expectLater.